sql/src/main/java/org/apache/druid/sql/calcite/planner/QueryHandler.java

```diff
@@ -563,7 +563,7 @@ protected PlannerResult planWithDruidConvention() throws ValidationException
                  .plus(DruidLogicalConvention.instance()),
           newRoot
       );
-      Hook.JAVA_PLAN.run(newRoot);
```

Comment:
I think it would be a bit nicer if this didn't have to rely on statics and could instead rely on an instance object.  For this `QueryHandler` itself, it looks like these end up constructed via the `DruidPlanner` which is built by the `PlannerFactory` which is injected via Guice.  Could we get the `DruidHook` (or something equivalent to it) injected into that and pushed down to this object instead of relying on statics?

Reply:
I was also thinking about doing that - but opted to not do that at first
fortunately 90% of the necessary machinery was already in place...so it wasn't really hard to plug it in via guice...but I had to touch a few more files (+20)